src/com/sittinglittleduck/DirBuster/Worker.java

```diff
@@ -573,20 +298,187 @@ else if(work.getBaseCaseObj().isUseRegexInstead())
+
+		Matcher m = regexFindFile.matcher(response);
+
+		//need to clean the base case of the item we are looking for
+		String basecase = FilterResponce.removeItemCheckedFor(work.getBaseCaseObj().getBaseCase(), work.getItemToCheck());
+
+		if(m.find())
+		{
+			System.out.println("DEBUG Worker[" + threadId + "]: 404 for: " + url.toString());
+		}
+		else if(!response.equalsIgnoreCase(basecase))
+		{
+			notifyItemFound(code, response, rawResponse, basecase);
+		}
+	}
+	
+	private void notifyItemFound(int code, String response, String rawResponse, String basecase, String type) {
+		if(work.isDir())
+		{
+		    if(Config.debug)
+		    {
+		        System.out.println("DEBUG Worker[" + threadId + "]: Found Dir (" + type +")" + url.toString());
+		    }
+		    //we found a dir
+		    manager.foundDir(url, code, response, basecase, rawResponse, work.getBaseCaseObj());
+		}
+		else
+		{
+		    //found a file
+		    if(Config.debug)
+		    {
+		        System.out.println("DEBUG Worker[" + threadId + "]: Found File (" + type +")" + url.toString());
+		    }
+		    manager.foundFile(url, code, response, basecase, rawResponse, work.getBaseCaseObj());
+		}
+	}
+	
+
+	private void notifyItemFound(int code, String response, String rawResponse, String basecase) {
+		notifyItemFound(code, response, rawResponse, basecase, "base case");		
+	}
+
+	private void parseHtml(HttpMethodBase httpMethod, String response) {
+		//parse the html of what we have found
+
+	    Header contentType = httpMethod.getResponseHeader("Content-Type");
+
+	    if(contentType != null)
+	    {
+	        if(contentType.getValue().startsWith("text"))
+	        {
+	            manager.addHTMLToParseQueue(new HTMLparseWorkUnit(response, work));
+	        }
+	    }
+		
+	}
+
+	private String getResponseAsString(HttpMethodBase httpMethod) throws IOException {
+		StringBuilder builder = new StringBuilder();
+		
+		Charset chartSet = getCharsetFrom(httpMethod);
+		
+		String result = new String(httpMethod.getResponseBody(), chartSet);
+		
+		try(BufferedReader input = new BufferedReader(new InputStreamReader(httpMethod.getResponseBodyAsStream()))) {
+			builder.append(System.lineSeparator());
+			builder.append(input.readLine());
+		}
+		return builder.toString();
+	}
+
+
+	private Charset getCharsetFrom(HttpMethodBase httpMethod) {
+		Charset chartSet;
+		
+		try {	
+			chartSet = Charset.forName(httpMethod.getRequestCharSet());
+		} catch (Exception ex) {
+			chartSet = Charset.forName("UTF-8");
+		}
+		return chartSet;
+	}
+
+	private String getHeadersAsString(HttpMethodBase httpMethod) {
+		Header[] headers = httpMethod.getResponseHeaders();
+		
+		StringBuilder builder = new StringBuilder(20 * (headers.length +1));
+		
+		builder.append(httpMethod.getStatusLine());
+		builder.append("\r\n");
+		
+		for (Header header : headers) {
+			builder.append(header.getName()).append(": ").append(header.getValue());
+			builder.append("\r\n");
+		}
+
+		return builder.toString();
```

Comment:
If the method `getResponseAsString()` no longer injects newline characters (by constructing the `String` with byte array), it should be appended another "\r\n" before calling `toString()`, it is necessary to properly separate the header from the body.